doc/book/zend.authentication.adapter.dbtable.md

```diff
@@ -0,0 +1,304 @@
+    )));
+
+    // store the identity as an object where the password column has
+    // been omitted
+    $storage->write($adapter->getResultRowObject(
+        null,
+        'password'
+    ));
+
+    /* ... */
+
+} else {
+
+    /* ... */
+
+}
+```
+
+### Advanced Usage By Example
+
+While the primary purpose of the `Zend\Authentication` component (and consequently
+`Zend\Authentication\Adapter\DbTable`) is primarily **authentication** and not **authorization**,
+there are a few instances and problems that toe the line between which domain they fit within.
+Depending on how you've decided to explain your problem, it sometimes makes sense to solve what
+could look like an authorization problem within the authentication adapter.
+
+With that disclaimer out of the way, `Zend\Authentication\Adapter\DbTable` has some built in
+mechanisms that can be leveraged for additional checks at authentication time to solve some common
+user problems.
+
+```php
+use Zend\Authentication\Adapter\DbTable as AuthAdapter;
+
+// The status field value of an account is not equal to "compromised"
+$adapter = new AuthAdapter($db,
+                           'users',
+                           'username',
+                           'password',
+                           'MD5(?) AND status != "compromised"'
+                           );
+
+// The active field value of an account is equal to "TRUE"
+$adapter = new AuthAdapter($db,
+                           'users',
+                           'username',
+                           'password',
+                           'MD5(?) AND active = "TRUE"'
+                           );
+```
+
+Another scenario can be the implementation of a salting mechanism. Salting is a term referring to a
+technique which can highly improve your application's security. It's based on the idea that
+concatenating a random string to every password makes it impossible to accomplish a successful brute
+force attack on the database using pre-computed hash values from a dictionary.
+
+Therefore, we need to modify our table to store our salt string:
+
+```php
+$sqlAlter = "ALTER TABLE [users] "
+          . "ADD COLUMN [password_salt] "
+          . "AFTER [password]";
+```
+
+Here's a simple way to generate a salt string for every user at registration:
+
+```php
+$dynamicSalt = '';
+for ($i = 0; $i < 50; $i++) {
+    $dynamicSalt .= chr(rand(33, 126));
+}
+```
+
+And now let's build the adapter:
+
+```php
+$adapter = new AuthAdapter($db,
+                           'users',
+                           'username',
+                           'password',
+                           "MD5(CONCAT('staticSalt', ?, password_salt))"
+                          );
+```
+
+> **note**
+>
+> You can improve security even more by using a static salt value hard coded into your application.
+In the case that your database is compromised (e. g. by an *SQL* injection attack) but your web
+server is intact your data is still unusable for the attacker.
```

Comment:
Same notes as for the initial note; let's get an improved title, and make sure all lines are prefixed with `> `.